src/pseudopeople/interface.py

```diff
@@ -50,61 +50,83 @@ def _generate_form(
-                f"{data_path.suffix}"
-            )
-        if not isinstance(data, pd.DataFrame):
-            raise TypeError(
-                f"File located at {data_path} must contain a pandas DataFrame. "
-                "Please provide the path to the unmodified root data directory."
-            )
-
-        # Coerce dtypes prior to noising to catch issues early as well as
-        # get most columns away from dtype 'category' and into 'object' (strings)
-        for col in columns_to_keep:
-            if col.dtype_name != data[col.name].dtype.name:
-                data[col.name] = data[col.name].astype(col.dtype_name)
+        data = _load_data_from_path(data_path, year_filter)
 
+        data = _coerce_dtypes(data, form)
+        data = _reformat_dates_for_noising(data)
         noised_data = noise_form(form, data, configuration_tree, seed)
-        noised_data = _extract_columns(columns_to_keep, noised_data)
+        noised_data = _extract_columns(form.columns, noised_data)
         noised_form.append(noised_data)
 
     noised_form = pd.concat(noised_form, ignore_index=True)
 
     # Known pandas bug: pd.concat does not preserve category dtypes so we coerce
     # again after concat (https://github.com/pandas-dev/pandas/issues/51362)
-    for col in columns_to_keep:
-        if col.dtype_name != noised_form[col.name].dtype.name:
-            noised_form[col.name] = noised_form[col.name].astype(col.dtype_name)
+    noised_form = _coerce_dtypes(noised_form, form)
 
     return noised_form
 
 
+def _coerce_dtypes(data: pd.DataFrame, form: Form):
+    # Coerce dtypes prior to noising to catch issues early as well as
+    # get most columns away from dtype 'category' and into 'object' (strings)
+    for col in form.columns:
+        if col.dtype_name != data[col.name].dtype.name:
+            data[col.name] = data[col.name].astype(col.dtype_name)
+    return data
+
+
+def _load_data_from_path(data_path: Path, year_filter: Dict):
+    """Load data from a data file given a data_path and a year_filter."""
+    if data_path.suffix == ".hdf":
+        with pd.HDFStore(str(data_path), mode="r") as hdf_store:
+            data = hdf_store.select("data", where=year_filter["hdf"])
+    elif data_path.suffix == ".parquet":
+        data = pq.read_table(data_path, filters=year_filter["parquet"]).to_pandas()
+    else:
+        raise ValueError(
+            "Source path must either be a .hdf or a .parquet file. Provided "
+            f"{data_path.suffix}"
+        )
+    if not isinstance(data, pd.DataFrame):
+        raise TypeError(
+            f"File located at {data_path} must contain a pandas DataFrame. "
+            "Please provide the path to the unmodified root data directory."
+        )
+    return data
+
+
+def _reformat_dates_for_noising(data: pd.DataFrame):
+    """Formats SSA event_date and dates of birth, so they can be noised."""
+    if COLUMNS.ssa_event_date.name in data.columns:
```

Comment:
If we happen to have a column named the same as the ssa event date column this will be applied to it whether we want to or not. I think it'd be better to pass the form through.

Reply:
That's true. Good point. I'll update for that.